docs/ONNXTypes.md

```diff
@@ -0,0 +1,65 @@
+## Optional Type
+
+An optional type represents a reference to either an element (could be Tensor, Sequence, Map, or Sparse Tensor) or a null value. The optional type appears in model inputs, outputs, as well as intermediate values.  
+
+### Use-cases
+
+Optional type enables users to represent more dynamic typing senarios in ONNX. Similar to Optional[X] type hint in Python typing which is equivalent to Union[None, X], Optional typs in ONNX may reference a single element, or null.
+
+### Examples in PyTorch
+Optional type only appears in TorchScript graphs generated by jit script compiler. Scripting a model captures dynamic types where an optional value can be assigned either None or a value. 
+
+- Example 1
```

Comment:
for every example can you add how the resulting onnx graph would look like with this type added... I know we dont have operator support yet but you can add a placeholder for these ops